core/che-core-api-core/src/main/java/org/eclipse/che/api/core/util/ProcessUtil.java

```diff
@@ -126,6 +130,83 @@ public static Process executeAndWait(
     return process;
   }
 
+  /**
+   * Start the process, writing the stdout of process to {@code stdout}, stderr of process to {@code
+   * stderr}, and terminate process by {@code timeout}.<br>
+   *
+   * @param commandLine arguments of process command
+   * @param timeout timeout for process. If process duration > {@code timeout} than kill process and
+   *     throw {@link TimeoutException}.
+   * @param timeUnit timeUnit of the {@code timeout}.
+   * @param stdout a consumer where stdout will be redirected
+   * @param stderr a consumer where stderr will be redirected
+   * @return the started process
+   * @throws InterruptedException in case terminate process
+   * @throws IOException in case I/O error, or command execution error, or stderr is not empty after
+   *     the command execution
+   * @throws TimeoutException if process gets more time then defined by {@code timeout}
+   */
+  public static Process executeAndWait(
+      String[] commandLine,
+      int timeout,
+      TimeUnit timeUnit,
+      ListLineConsumer stdout,
+      ListLineConsumer stderr)
+      throws TimeoutException, IOException, InterruptedException {
```

Comment:
All methods of this class are using LineConsumer why you limit it with ListLineConsumer?

Reply:
The problem is [LineConsumer](https://github.com/eclipse/che/blob/master/core/che-core-api-core/src/main/java/org/eclipse/che/api/core/util/LineConsumer.java) interface doesn't have methods to read consumed line, whereas [ListLineConsumer](https://github.com/eclipse/che/blob/master/core/che-core-api-core/src/main/java/org/eclipse/che/api/core/util/ListLineConsumer.java) has. So, usage of **ListLineConsumer** simplifies reading **stderr** after execution of command in code `String errorMessage = stderr.getText();`.
Of course, the method can be re-wrote to use **LineConsumer** arguments, but it will add potentially unnecessary complexity to it, IMHO.

Reply:
What is the main purpose of ```executeAndWait``` to execute and wait or to show the error message? Are you sure you need to read ALL message from stderror to send part of it in an exception?

Reply:
One more question. What should caller method do with Process in case if it's not completed during the desired timeout?

Reply:
In case if you change ListLineConsumer  to LineConsumer I see no reason to limit it.

Reply:
Okay, I will change it to avoid OOM.